### PR TITLE
Paging refactor

### PR DESCRIPTION
**Pull request by ambercs**
Refactor paging of v3 api objects to use the same url parsing function.
Also discovered that listing v3 isolation segments was not working properly and added a fix.

### CODE REVIEW

**Comment by ambercs** [comment]
@jghiloni @zmb3

File: isolationsegments.go [hunk 1]
```diff
@@ -100,10 +100,13 @@ func (c *Client) GetIsolationSegmentByGUID(guid string) (*IsolationSegment, erro
 
 func (c *Client) ListIsolationSegmentsByQuery(query url.Values) ([]IsolationSegment, error) {
 	var iss []IsolationSegment
-	requestUrl := "/v3/isolation_segments?" + query.Encode()
+	requestURL := "/v3/isolation_segments" + query.Encode()
```
Comment (zmb3) [inline isolationsegments.go:103]: You're appending the query string twice here.

**Review by zmb3 — CHANGES_REQUESTED** [review]

File: isolationsegments.go (hunk 1 above)
Comment (ambercs) [reply isolationsegments.go]: good catch, thanks!

**Comment by zmb3** [comment]
Looks great, thanks!

**Pull request merged**